packages/client/src/client.ts

```diff
@@ -16,13 +17,25 @@ export class Client {
   static *create(url: string): Operation<Client> {
     let socket = new w3cwebsocket(url) as WebSocket;
 
+    yield spawn(function* detectStartupError(): Operation<void> {
+      let [error] = yield once(socket, 'error');
+      
+      if (isErrorEvent(error)) {
+        throw new NoServerError(`Could not connect to server at ${url}`);
+      } else {
+        throw error;
```

Comment:
I'm trying to imagine a scenario where we execute this code branch, and can't think of one other than a bug in the `websocket` library. Still, it's good to have in case there is and it will throw a big nasty stack trace.